client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -422,8 +449,13 @@ public void flush() throws SegmentSealedException {
     }
 
     @Override
-    public Collection<PendingEvent> getUnackedEvents() {
-        return Collections.unmodifiableCollection(state.getAllInflightEvents());
-    }
+    public List<PendingEvent> seal() {
+        // Seal function is invoked by SegmentSealedCallback. i.e. when SegmentSealedCallback / Seal function is
+        // invoked there are no writes happening to the Segment.
 
+        //close connection and update the exception to SegmentSealed, this ensures further writes receive a
+        // SegmentSealedException.
+        state.failConnection(new SegmentSealedException(this.segmentName));
```

Comment:
We shouldn't but that is only because we know the behavior of the caller. Each class must be correct on its own. See my comment above for a fix.